src/offline/syncUpdatesWithBackend.ts

```diff
@@ -2,28 +2,175 @@ import { ProcosysApiService } from '../services/procosysApi';
+            return;
+        }
+
+        //Response is ok
+        //Set offline update to be syncronized
+        offlineUpdate.syncronized = true;
+        offlineUpdateRepository.updateOfflineUpdate(offlineUpdate);
+
+        if (offlineUpdate.responseIsNewEntityId) {
+            //todo: Her kan vi ha litt feilhådntering. Vi må ha en reposnse.id her.
+            console.log('offlineupadte har new entity id: ', newEntityId);
+            return newEntityId;
+        }
+    };
+
+    /**
+     * Update entityId for remaining updates for entity.
+     * This is necessary when backend has generated a new id for the entity (a new entity was created offline).
+     * The specific PUT operation must be identified, and the specific variable in the body must be updated.
+     */
+    const updateEntityId = (
+        newEntityId: number,
+        offlineUpdate: OfflineUpdateRequest
+    ): void => {
+        const method = offlineUpdate.method.toUpperCase();
+
+        if (method == 'PUT') {
+            if (offlineUpdate.url.startsWith('PunchListItem/')) {
+                //putUpdatePunch
+                offlineUpdate.bodyData.PunchItemId = newEntityId.toString();
+                return;
+            }
+        }
+
+        if (method == 'POST') {
+            if (
+                offlineUpdate.url.startsWith('CheckList/CustomItem/SetOk') ||
+                offlineUpdate.url.startsWith('CheckList/CustomItem/Clear')
+            ) {
+                //postCustomClear and postCustomSetOK
+                offlineUpdate.bodyData.CustomCheckItemId =
+                    newEntityId.toString();
+                return;
+            }
+        }
+
+        console.error(
+            'Not able to update entityId for PUT (new entity where new id has been generated by server).',
+            offlineUpdate.url
+        );
+    };
+
+    //----------------------------------------
+    // Syncronize all updates with backend.
+    //----------------------------------------
     console.log('SyncUpdatesWithBackend');
     const offlineUpdateRepository = new OfflineUpdateRepository();
-    const allOfflineUpdates = await offlineUpdateRepository.getAllRequests();
-
-    allOfflineUpdates.forEach((updateRequest: OfflineUpdateRequest) => {
-        console.log('syncUpdatesWithBackend: ', updateRequest);
-        if (updateRequest.method.toLowerCase() == 'post') {
-            if (updateRequest.type == RequestType.Json) {
-                api.postByFetch(updateRequest.url, updateRequest.bodyData);
-            } else if (updateRequest.type == RequestType.Attachment) {
-                api.postAttachmentByFetch(
-                    updateRequest.url,
-                    updateRequest.bodyData
+
+    const offlineUpdates =
+        await offlineUpdateRepository.getRequestsOrderedByEntityAndSeq();
+
+    //Get a distinct set of entities to syncronize
+    const entitiesToUpdate = new Set<number>();
+    for (const offlineUpdate of offlineUpdates) {
```

Comment:
We usually avoid for loops as often as possible. Try to use offlineUpdates.forEact or another array method